### ISSUE TITLE
View query triggered by active-only changes results in incomplete channel cache

### ISSUE THREAD

**Issue by adamcfraser** [couchbase/sync_gateway#2955]
When a channel cache is initialized by an active-only changes feed, it appears to be only adding the active-only results to the channel cache.  Subsequent non-active-only changes requests are missing deleted/removed documents.

Reported when testing against a walrus bucket as part of client unit tests.

**Comment by adamcfraser** [comment]
Reproduced the same behaviour when running against Couchbase Server:

1. Create a document
2. Delete the document
3. Restart Sync Gateway (to force rebuild of cache from view)
4. Issue an active_only=true changes request
5. Issue an active_only=false changes request

The deleted document is not included in the result set in step 5.

**Comment by adamcfraser** [comment]
This appears to be a regression in 1.5 introduced by https://github.com/couchbase/sync_gateway/pull/2805

**Comment by raghusarangapani** [comment]
I wrote a functional test using the steps mentioned above and tested it out against vagrant VMs locally with 1.5.0-589 and I see a panic at step 4:

```
2017-10-05T21:13:23.976Z HTTP:  #001: GET /db/_changes?active_only=true  (ADMIN)
2017-10-05T21:13:23.976Z Changes+: Int sequence multi changes feed...
2017-10-05T21:13:23.976Z Changes: MultiChangesFeed(channels: {*}, options: {Since:0 Limit:0 Conflicts:false IncludeDocs:false Wait:false Continuous:false Terminator:0xc42021ab40 HeartbeatMs:0 TimeoutMs:300000 ActiveOnly:true}) ... 
2017-10-05T21:13:23.976Z Changes+: MultiChangesFeed: channels expand to "" ... 
2017-10-05T21:13:23.976Z Cache: Initialized cache for channel "*" with options: &{ChannelCacheMinLength:50 ChannelCacheMaxLength:500 ChannelCacheAge:1m0s}
2017-10-05T21:13:23.976Z Cache: getCachedChanges("*", 0) --> 0 changes valid from #7
2017-10-05T21:13:23.976Z Cache:   Querying 'channels' view for "*" (start=#1, end=#7, limit=0)
2017-10-05T21:13:23.992Z Changes: MultiChangesFeed done 
panic: runtime error: index out of range

goroutine 69 [running]:
panic(0xb26820, 0xc4200100a0)
	/usr/local/go/1.7.4/go/src/runtime/panic.go:500 +0x1a1
github.com/couchbase/sync_gateway/db.(*DatabaseContext).getChangesInChannelFromView(0xc420334000, 0xbef0f1, 0x1, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/couchbase/jenkins/workspace/sgw-unix-build@2/1.5.0/enterprise/godeps/src/github.com/couchbase/sync_gateway/db/changes_view.go:88 +0xeda
github.com/couchbase/sync_gateway/db.(*channelCache).GetChanges(0xc42021e820, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/couchbase/jenkins/workspace/sgw-unix-build@2/1.5.0/enterprise/godeps/src/github.com/couchbase/sync_gateway/db/channel_cache.go:224 +0x497
github.com/couchbase/sync_gateway/db.(*changeCache).GetChanges(0xc4200d4a50, 0xbef0f1, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/couchbase/jenkins/workspace/sgw-unix-build@2/1.5.0/enterprise/godeps/src/github.com/couchbase/sync_gateway/db/change_cache.go:730 +0x133
github.com/couchbase/sync_gateway/db.(*Database).changesFeed(0xc4203a2020, 0xbef0f1, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/couchbase/jenkins/workspace/sgw-unix-build@2/1.5.0/enterprise/godeps/src/github.com/couchbase/sync_gateway/db/changes.go:150 +0x105
github.com/couchbase/sync_gateway/db.(*Database).SimpleMultiChangesFeed.func1(0x0, 0x0, 0xc42021acc0, 0xc4203a2020, 0xc42021e460, 0xc420363ef0)
	/home/couchbase/jenkins/workspace/sgw-unix-build@2/1.5.0/enterprise/godeps/src/github.com/couchbase/sync_gateway/db/changes.go:463 +0x6e2
created by github.com/couchbase/sync_gateway/db.(*Database).SimpleMultiChangesFeed
	/home/couchbase/jenkins/workspace/sgw-unix-build@2/1.5.0/enterprise/godeps/src/github.com/couchbase/sync_gateway/db/changes.go:674 +0x253
```

**Comment by tleyden** [comment]
@raghusarangapani can you post a link to the functional test?

**Comment by raghusarangapani** [comment]
It is in my branch:
https://github.com/couchbaselabs/mobile-testkit/blob/feature/changes/testsuites/syncgateway/functional/tests/test_changes.py

**Comment by adamcfraser** [comment]
Fixed by #2956.
